core/benches/bench.rs

```diff
@@ -0,0 +1,732 @@
+        total_bytes: usize,
+        concurrency: u32,
+    }
+
+    impl BenchBuilder {
+        pub fn new(name: &str) -> Self {
+            Self {
+                name: name.to_string(),
+                disk_type: None,
+                io_type: None,
+                io_pattern: None,
+                buf_size: 4 * KiB,
+                total_bytes: 1 * MiB,
+                concurrency: 1,
+            }
+        }
+
+        pub fn disk_type(mut self, disk_type: DiskType) -> Self {
+            self.disk_type = Some(disk_type);
+            self
+        }
+
+        pub fn io_type(mut self, io_type: IoType) -> Self {
+            self.io_type = Some(io_type);
+            self
+        }
+
+        pub fn io_pattern(mut self, io_pattern: IoPattern) -> Self {
+            self.io_pattern = Some(io_pattern);
+            self
+        }
+
+        pub fn buf_size(mut self, buf_size: usize) -> Self {
+            self.buf_size = buf_size;
+            self
+        }
+
+        pub fn total_bytes(mut self, total_bytes: usize) -> Self {
+            self.total_bytes = total_bytes;
+            self
+        }
+
+        pub fn concurrency(mut self, concurrency: u32) -> Self {
+            self.concurrency = concurrency;
+            self
+        }
+
+        pub fn build(self) -> Result<Box<dyn Bench>> {
+            let Self {
+                name,
+                disk_type,
+                io_type,
+                io_pattern,
+                buf_size,
+                total_bytes,
+                concurrency,
+            } = self;
+
+            let disk_type = match disk_type {
+                Some(disk_type) => disk_type,
+                None => return_errno_with_msg!(Errno::InvalidArgs, "disk_type is not given"),
+            };
+            let io_type = match io_type {
+                Some(io_type) => io_type,
+                None => return_errno_with_msg!(Errno::InvalidArgs, "io_type is not given"),
+            };
+            let io_pattern = match io_pattern {
+                Some(io_pattern) => io_pattern,
+                None => return_errno_with_msg!(Errno::InvalidArgs, "io_pattern is not given"),
+            };
+            if total_bytes == 0 || total_bytes % BLOCK_SIZE != 0 {
+                return_errno_with_msg!(
+                    Errno::InvalidArgs,
+                    "total_bytes must be greater than 0 and a multiple of block size"
+                );
+            }
+            if buf_size == 0 || buf_size % BLOCK_SIZE != 0 {
+                return_errno_with_msg!(
+                    Errno::InvalidArgs,
+                    "buf_size must be greater than 0 and a multiple of block size"
+                );
+            }
+            if concurrency == 0 {
+                return_errno_with_msg!(Errno::InvalidArgs, "concurrency must be greater than 0");
+            }
+
+            Ok(Box::new(SimpleDiskBench {
+                name,
+                disk_type,
+                disk: None,
```

Comment:
We could construct `disk` above here, rather than within `SimpleDiskBench::prepare_disk`.